### ISSUE TITLE
Why GPT2ForTokenClassification doesn't shift logits and labels?

### ISSUE THREAD

**Issue by JaheimLee** [huggingface/transformers#14323]
Why GPT2ForTokenClassification doesn't shift logits and labels? I think it's loss is similar to  GPT2LMHeadModel. By the way, the docstring of GPT2ForTokenClassification in forward function seems not right!

**Comment by NielsRogge** [comment]
Hi,

`GPT2ForTokenClassification` behaves exactly as `BertForTokenClassification`. There's no need to shift the labels; it just needs to predict a label for every token.

`GPT2LMHeadModel` on the other hand needs to predict the next token given the previous ones, which is why the logits and labels are shifted.

**Comment by JaheimLee** [comment]
Got it! Thanks for your reply.
